ctest-next/src/translator.rs

```diff
@@ -0,0 +1,319 @@
+                    .replacen("(*)", &format!("(*{modifier})"), 1)
+                    .trim()
+                    .to_string())
+            }
+            syn::Type::Reference(_) | syn::Type::Ptr(_) => {
+                let inner_type = self.translate_type(ptr.elem.deref())?;
+                if inner_type.contains("(*)") {
+                    Ok(inner_type
+                        .replacen("(*)", &format!("(*{modifier} *)"), 1)
+                        .trim()
+                        .to_string())
+                } else {
+                    Ok(format!("{inner_type} {modifier}*").trim().to_string())
+                }
+            }
+            _ => {
+                let inner_type = self.translate_type(inner)?;
+                Ok(format!("{inner_type} {modifier}*"))
+            }
+        }
+    }
+
+    /// Translate a simple Rust expression to C.
+    #[expect(clippy::only_used_in_recursion)]
+    fn translate_expr(&self, expr: &syn::Expr) -> Result<String, TranslationError> {
+        match expr {
+            syn::Expr::Lit(l) => match &l.lit {
+                syn::Lit::Int(i) => Ok(i.to_string()),
+                lit => Err(TranslationError::InvalidSyntax(
+                    lit.to_token_stream().to_string(),
+                )),
+            },
+            syn::Expr::Path(p) => Ok(p.path.segments.last().unwrap().ident.to_string()),
+            syn::Expr::Cast(c) => self.translate_expr(c.expr.deref()),
+            syn::Expr::Paren(p) => {
+                let inner = self.translate_expr(&p.expr)?;
+                Ok(format!("({inner})"))
+            }
+            syn::Expr::Group(g) => self.translate_expr(&g.expr),
+            syn::Expr::Binary(b) => {
+                let left = self.translate_expr(b.left.deref())?;
+                let right = self.translate_expr(b.right.deref())?;
+
+                match b.op {
+                    syn::BinOp::Add(_) => Ok(format!("{left} + {right}")),
+                    syn::BinOp::Sub(_) => Ok(format!("{left} - {right}")),
+                    syn::BinOp::Mul(_) => Ok(format!("{left} * {right}")),
+                    syn::BinOp::Div(_) => Ok(format!("{left} / {right}")),
+                    syn::BinOp::Rem(_) => Ok(format!("{left} % {right}")),
+                    syn::BinOp::Shl(_) => Ok(format!("{left} << {right}")),
+                    syn::BinOp::Shr(_) => Ok(format!("{left} >> {right}")),
+                    syn::BinOp::BitAnd(_) => Ok(format!("{left} & {right}")),
+                    syn::BinOp::BitOr(_) => Ok(format!("{left} | {right}")),
+                    syn::BinOp::BitXor(_) => Ok(format!("{left} ^ {right}")),
+                    _ => Err(TranslationError::UnknownOperator(
+                        b.op.to_token_stream().to_string(),
+                    )),
+                }
+            }
+            expr => Err(TranslationError::UnknownExpression(
+                expr.to_token_stream().to_string(),
+            )),
+        }
```

Comment:
Could we just leave expressions 1:1, i.e. only do `to_string()`? Most should map pretty directly, you shouldn't have to handle binops manually.

If anything, give it a regex pass and delete `(\B::)?\w+::` to get rid of paths.